gomoji.go

```diff
@@ -82,6 +82,46 @@ func GetInfo(emoji string) (Emoji, error) {
 	return em, nil
 }
 
+// CollectAll finds all emojis in given string. Unlike FindAll, this does not
+// distinct repeating occurrences of emoji. If there are no emojis it returns a nil-slice.
+func CollectAll(s string) []Emoji {
+	var emojis []Emoji
+
+	lastSeen := 0
+
+	gr := uniseg.NewGraphemes(s)
+	for gr.Next() {
+
+		start, end := gr.Positions()
+
+		for _, r := range s[lastSeen:start] {
+			if em, ok := emojiMap[string(r)]; ok {
+				emojis = append(emojis, em)
+			}
+		}
+
+		if em, ok := emojiMap[gr.Str()]; ok {
+			emojis = append(emojis, em)
+		} else {
+			for _, r := range s[start:end] {
+				if em, ok := emojiMap[string(r)]; ok {
+					emojis = append(emojis, em)
+				}
+			}
+		}
+
+		lastSeen = end
+	}
```

Comment:
The cycle looks hard to get, though. 
Maybe we can try something like this? Will it work for you? It looks like it passes your test cases.

```
func CollectAll(s string) []Emoji {
	var emojis []Emoji

	gr := uniseg.NewGraphemes(s)
	for gr.Next() {
		if em, ok := emojiMap[gr.Str()]; ok {
			emojis = append(emojis, em)
			continue
		}

		start, end := gr.Positions()
		for _, r := range s[start:end] {
			if em, ok := emojiMap[string(r)]; ok {
				emojis = append(emojis, em)
			}
		}
	}

	return emojis
}
```